### PR TITLE
Allow checks to opt-out of last run results save

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Allow checks to opt-out of last run results save. Set the `connections` check to opt-out.

### Motivation

The `connections` check results can be quite large, and having the last results in a flare is not valuable to the NPM team.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by nammn — COMMENTED** [review]
I think you are missing the "normal" pod check: https://github.com/DataDog/datadog-agent/blob/c90c24ffdbb1f66049a069d5f3b05191f42325bf/pkg/process/checks/pod.go


`pkg/process/checks/pod.go`

File: pkg/process/checks/process.go [hunk 1]
```diff
@@ -18,8 +18,6 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/process/util"
 	"github.com/DataDog/datadog-agent/pkg/util/cloudproviders"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
-	"github.com/DataDog/gopsutil/cpu"
-	"go.uber.org/atomic"
```
Comment (robertjli) [inline pkg/process/checks/process.go:22]: Looks like this was removed unintentionally?

**Review by robertjli — APPROVED** [review]
Change looks good to me, but I'm not sure what's going on with some of the failing Github checks..

File: pkg/process/checks/process.go (hunk 1 above)
Comment (brycekahle) [reply pkg/process/checks/process.go:22]: Yes, adding back in.

**Review by nammn — APPROVED** [review]
lgtm for the pod check

**Review by hmahmood — APPROVED** [review]

**Pull request merged**